### ISSUE TITLE
exp function with percent! argument gives wrong result

### ISSUE THREAD

**Issue by meijeru** [red/red#2699]
`exp`  has a `number!` argument, thus `percent!` should be allowed.
```
>> exp 100%
== 1.0    ;; <--- should be e = 2.718281828459045
```
I think that all mathematical actions could be usefully reviewed for checking for TYPE_PERCENT in addition to TYPE_FLOAT

**Comment by meijeru** [comment]
In this case, it seems to be `argument-as-float` in `%runtime/natives.reds`

**Comment by greggirwin** [comment]
Does it make sense to support percent values with `exp`, or should the type spec be changed instead?

**Comment by dockimbel** [comment]
We should not support percent! for trigonometric functions.

**Comment by greggirwin** [comment]
See also: https://github.com/red/red/issues/2586

**Comment by dockimbel** [comment]
Fixed by the fix for #2586.
